core/src/main/java/com/linecorp/armeria/common/stream/StreamMessageWrapper.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.stream;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+import org.reactivestreams.Subscriber;
+
+/**
+ * Wraps a {@link StreamMessage} and forwards invocations of methods to {@code delegate}.
+ * @param <T> the type of elements
+ */
+public class StreamMessageWrapper<T> implements StreamMessage<T> {
+
+    private final StreamMessage<? extends T> delegate;
```

Comment:
How about adding a protected getter method that returns the delegate? e.g. `protected StreamMessage<? extends T> delegate()`